src/scanpy/preprocessing/_qc.py

```diff
@@ -191,21 +194,17 @@ def describe_var(
         if issparse(X):
             X.eliminate_zeros()
     var_metrics = pd.DataFrame(index=adata.var_names)
-    if issparse(X):
-        # Current memory bottleneck for csr matrices:
-        var_metrics["n_cells_by_{expr_type}"] = X.getnnz(axis=0)
-        var_metrics["mean_{expr_type}"] = mean_variance_axis(X, axis=0)[0]
-    else:
-        var_metrics["n_cells_by_{expr_type}"] = np.count_nonzero(X, axis=0)
-        var_metrics["mean_{expr_type}"] = X.mean(axis=0)
+    var_metrics["n_cells_by_{expr_type}"], var_metrics["mean_{expr_type}"] = (
```

Comment:
I’m just realizing that none of these are `f`-literals. `expr_type` won’t be interpolated into this!

mayyybe we should fix that.

Reply:
i think they are! `col.format` is called!

Reply:
ah! bizarre! but sure, let’s do that in another PR then